subsys/bluetooth/host/gatt.c

```diff
@@ -493,19 +604,22 @@ static void remove_cf_cfg(struct bt_conn *conn)
 	}
 }
 #endif /* CONFIG_BT_GATT_CACHING */
-#endif /* CONFIG_BT_GATT_DYNAMIC_DB */
 
 BT_GATT_SERVICE_DEFINE(_1_gatt_svc,
 	BT_GATT_PRIMARY_SERVICE(BT_UUID_GATT),
-#if defined(CONFIG_BT_GATT_DYNAMIC_DB)
+#if defined(CONFIG_BT_GATT_SERVICE_CHANGED)
 	/* Bluetooth 5.0, Vol3 Part G:
 	 * The Service Changed characteristic Attribute Handle on the server
 	 * shall not change if the server has a trusted relationship with any
 	 * client.
 	 */
 	BT_GATT_CHARACTERISTIC(BT_UUID_GATT_SC, BT_GATT_CHRC_INDICATE,
 			       BT_GATT_PERM_NONE, NULL, NULL, NULL),
-	BT_GATT_CCC(sc_ccc_cfg_changed, BT_GATT_PERM_READ | BT_GATT_PERM_WRITE),
+	BT_GATT_CCC_MANAGED((&(struct _bt_gatt_ccc)
```

Comment:
Maybe make this a named struct, and then we can compare ccc with that struct instead of checking if the cfg_changed function is sc_ccc_cfg_changed. (Line 1680)

Reply:
Done in latest push!